pkg/apis/mysql/v1alpha1/mysqlcluster_types.go

```diff
@@ -101,6 +101,9 @@ type MysqlClusterSpec struct {
 	// +optional
 	VolumeSpec VolumeSpec `json:"volumeSpec,omitempty"`
 
+	// Master service extra specifiaction
+	MasterServiceSpec ServiceSpec `json:"masterServiceSpec,omitempty"`
```

Comment:
You can use as type directly [`*core.ServiceSpec`](https://godoc.org/k8s.io/api/core/v1#ServiceSpec) so a user can easely specify more fields if needed. And in the object syncer only override the selector and internal port,if `masterServiceSpec` is specified.